server/src/main/java/keywhiz/service/daos/SecretContentDAO.java

```diff
@@ -74,12 +78,39 @@ public long createSecretContent(long secretId, String encryptedContent,
     r.setExpiry(expiry);
     r.store();
 
+    pruneOldContents(secretId);
```

Comment:
A combination of both might be cool?  So keep at least 10 secrets, and 45 days.  Then you don't risk automation gone crazy wiping out a secret by creating 11 bad versions -- as unlikely as that is.